src/models/financialBankruptcy.js

```diff
@@ -0,0 +1,81 @@
+import name from 'models/shared/name'
+import address from 'models/shared/locations/address'
+import { hasYesOrNo } from 'models/validate'
+
+const petitionTypes = [
+  'Chapter7',
+  'Chapter11',
+  'Chapter12',
+  'Chapter13',
+]
+
+const financialBankruptcy = {
+  PetitionType: {
+    presence: true,
+    hasValue: {
+      validator: {
+        inclusion: petitionTypes,
+      },
+    },
+  },
+  CourtAddress: {
+    presence: true,
+    location: { validator: address },
+  },
+  CourtInvolved: { presence: true, hasValue: true },
+  CourtNumber: { presence: true, hasValue: true },
+  TotalAmount: {
+    presence: true,
+    hasValue: {
+      validator: { numericality: true },
+    },
+  },
+  DateFiled: {
+    presence: true,
+    date: { requireDay: false },
+  },
+  DateDischarged: (value, attributes) => {
+    const { DateDischargedNotApplicable } = attributes
+    if (DateDischargedNotApplicable && DateDischargedNotApplicable.applicable) {
+      return {
+        presence: true,
+        date: { requireDay: false },
+      }
+    }
+    return {}
+  },
+  NameDebt: {
+    presence: true,
+    model: { validator: name },
+  },
+  HasDischargeExplanation: {
+    presence: true,
+    hasValue: { validator: hasYesOrNo },
+  },
+  DischargeExplanation: (value, attributes) => {
+    const { HasDischargeExplanation } = attributes
+    if (HasDischargeExplanation && HasDischargeExplanation.value === 'Yes') {
```

Comment:
just looking at the existing validator and the form, it seems like DischargeExplanation is required whether HasDischargeExplanation is yes or no